src/io/GeoJSON.cpp

```diff
@@ -0,0 +1,124 @@
+/**********************************************************************
+ *
+ * GEOS - Geometry Engine Open Source
+ * http://geos.osgeo.org
+ *
+ * Copyright (C) 2005-2006 Refractions Research Inc.
+ *
+ * This is free software; you can redistribute and/or modify it under
+ * the terms of the GNU Lesser General Public Licence as published
+ * by the Free Software Foundation.
+ * See the COPYING file for more information.
+ *
+ **********************************************************************/
+
+#include <geos/io/GeoJSON.h>
+
+#undef DEBUG_GEOJSON
+
+namespace geos {
+namespace io { // geos.io
+
+// GeoJSONValue
+
+GeoJSONValue GeoJSONValue::createStringValue(std::string s) {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::STRING;
+    value.stringValue = s;
+    return value;
+}
+
+GeoJSONValue GeoJSONValue::createNumberValue(double n) {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::NUMBER;
+    value.numberValue = n;
+    return value;
+}
+
+GeoJSONValue GeoJSONValue::createNullValue() {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::NULLTYPE;
+    value.nullValue = nullptr;
+    return value;
+}
+
+GeoJSONValue GeoJSONValue::createBooleanValue(bool b) {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::BOOLEAN;
+    value.booleanValue = b;
+    return value;
+}   
+
+GeoJSONValue GeoJSONValue::createObjectValue() {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::OBJECT;
+    return value;
+}
+
+GeoJSONValue GeoJSONValue::createArrayValue() {
+    GeoJSONValue value{};
+    value.type = GeoJSONValueType::ARRAY;
+    return value;
+}
+
+std::string GeoJSONValue::toString() const {
+    std::stringstream os;
```

Comment:
I would tend to reverse this, have the << operator do all the work and have toString() just call the operator. Allows the << operator to stream if necessary/possible.

Reply:
When I refactored GeoJSONValue to use a union, I got rid of this.  I can add it back it you want.